buildbot/buildbot/worker.py

```diff
@@ -375,10 +399,66 @@ def stage_hls(db, config):
             )
 
 
+def stage_afi(db, config):
+    """AFI stage: Create the AWS FPGA binary and AFI from the *.xclbin 
+    (Xilinx FPGA binary file)
+    """
+    with work(db, state.AFI_START, state.AFI, state.HLS_FINISH) as task:
+        # clean all the generated files from previous runs.
+        task.run(
+            ['sh', '-c', 'cd xclbin; \
+              rm -rf to_aws; rm -rf *afi_id.txt;\
+              rm -rf *.tar; rm -rf *afgi_id.txt'],
+            cwd=CODE_DIR
+        )
+        # find the hw synthesis xclbin file.
+        xclbin_file_path = task.run(
+            ['sh', '-c', 'find ./xclbin  -type f -iname "*hw\.*.xclbin"'],
```

Comment:
This probably doesn't need a shell wrapper. You can just do `['find', './xclbin', '-type', '-f', ...]`. Or, it may even be possible to find the file through "normal means" without even executing a shell by using Python's [glob](https://docs.python.org/3/library/glob.html) module.